api/src/libs/ssmUtils.ts

```diff
@@ -0,0 +1,52 @@
+import { GetParameterCommand, PutParameterCommand, SSMClient } from "@aws-sdk/client-ssm";
+
+const ssmClient = new SSMClient({});
+const parameterName = `/${process.env.STAGE}/feature-flag/users-migration/kill-switch`;
+
+let cache: {
+  value?: boolean;
+  expiresAt?: number;
+} = {};
+
+export const isKillSwitchOn = async (): Promise<boolean> => {
+  const now = Date.now();
+
+  if (cache.value !== undefined && cache.expiresAt && cache.expiresAt > now) {
+    return cache.value;
+  }
+  try {
+    const command = new GetParameterCommand({
+      Name: parameterName,
+      WithDecryption: true,
+    });
+
+    const response = await ssmClient.send(command);
+    const paramValue = response.Parameter?.Value ?? "false";
+    const boolValue = paramValue.toLowerCase() === "true";
+
+    cache = {
+      value: boolValue,
+      expiresAt: now + 30 * 1000,
+    };
+
+    return boolValue;
+  } catch (error) {
+    console.error("Failed to read kill switch parameter:", error);
+    return false;
+  }
+};
+
+export const updateKillSwitch = async (value: boolean): Promise<void> => {
```

Comment:
[pebble] this function is only being called with the value `true`, so it's only purpose is to turn on the kill switch, thoughts on changing it to represent that if that's the only way you anticipate using this function, so no parameter to pass in and the function renamed to something like `turnOnKillSwitch`